src/App.tsx

```diff
@@ -9,37 +9,27 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
 import { Post } from './types/Post';
+import { useAppSelector } from './hooks/useAppSelector.hook';
+import { useAppDispatch } from './hooks/useAppDispatch.hook';
+import { fetchPosts } from './slices/posts.slice';
+import { resetCurrentPost, setCurrentPost } from './slices/selectedPost.slice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { author } = useAppSelector(state => state.author);
+  const { post } = useAppSelector(state => state.selectedPost);
+  const { items, hasError, loaded } = useAppSelector(state => state.posts);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const setSelectedPost = (newPost: Post | null) => {
+    dispatch(setCurrentPost(newPost));
+  };
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(resetCurrentPost());
 
     if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
+      dispatch(fetchPosts(author?.id));
```

Comment:
The optional chaining operator `?.` is not necessary here since `author` is already checked for truthiness in the `if` condition. You can safely use `author.id`.